### PR TITLE
[FLINK-9874][E2E Tests] Fix set_ssl_conf for macOS

### PR DESCRIPTION
**Pull request by florianschmidt1994**
## What is the purpose of the change

This fixes the set_ssl_conf utility function under macOS. The previous
version was using `hostname -I` regardless of the OS, but the -I option
is not available on the BSD version of the hostname command.

This is now fixed by checking for all IPv4 addresses from ifconfig if the
OS is macOS and formatting the output to be identical to `hostname -I`.

Additionally the filtering of the output is removed so that now all
ip addresses are appended to the SANSTRING instead of just one.

## Verifying this change
- Successfully ran `./run-single-test.sh test-scripts/test_batch_allround.sh` (this uses SSL setup) on macOS 10.13.6 
- Successfully ran `./run-single-test.sh test-scripts/test_batch_allround.sh` (this uses SSL setup) on Ubuntu 16.04.4

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by florianschmidt1994** [comment]
@NicoK here we go 🙌

**Comment by NicoK** [comment]
just out of curiosity (and I don't know the details there):
Does this still work with the gnu versions of these tools installed, e.g. via `homebrew`? Because at the moment, this completely depends on the OS detection, not on the actual tools.

**Comment by florianschmidt1994** [comment]
According to [1]`ifconfig` is not part of coreutils, but more like "a wrapper to the IPConfiguration agent". IDK if there is any other version of ifconfig out there that could be installed on MacOS, but if it were it would possibly break the code.. I'd say the risk of that happening is rather slim though

[1] https://en.wikipedia.org/wiki/List_of_GNU_Core_Utilities_commands
[2] https://en.wikipedia.org/wiki/Ifconfig

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 1]
```diff
@@ -154,12 +175,15 @@ function set_conf_ssl {
        rm -rf "${TEST_DATA_DIR}/ssl"
     fi
     mkdir -p "${TEST_DATA_DIR}/ssl"
+
     NODENAME=`hostname -f`
     SANSTRING="dns:${NODENAME}"
-    for NODEIP in `hostname -I | cut -d' ' -f1` ; do
+    for NODEIP in $(get_node_ip) ; do
```
Comment (twalthr) [inline flink-end-to-end-tests/test-scripts/common.sh]: Why did you remove the additional `| cut -d' ' -f1`?

Comment (florianschmidt1994) [reply flink-end-to-end-tests/test-scripts/common.sh]: `hostname -I ` would return a list of IP addresses separated by a space, and the cut would only select one of those. The intended behaviour according to my discussion with @NicoK is that we append all IP addresses returned by `hostname -I` to the SANSTRING, which is why I removed that line

Comment (twalthr) [reply flink-end-to-end-tests/test-scripts/common.sh]: Thanks for the clarification.
